### DISCUSSION TITLE
MSI Mutex checking when installing/uninstalling MSI?

### DISCUSSION THREAD

**Discussion by mgroover** [PSAppDeployToolkit/PSAppDeployToolkit#656] [Q&A] [answered]
From what I could see in the source, the only place where the Mutex lock function ("Test-IsMutexAvailable") is used and available as an option is in the "Execute-Process" function call.

Is there a specific reason why this hasn't been implemented in the Execute-MSI section? It would most definitely minimize the amount of deployment errors you get in larger environments where there is bound to be rogue installations running.

**Comment by That-Annoying-Guy** [comment, marked as answer]
BTW: Execute-MSI calls Execute-Process and it does check for Mutex lock.

For new packagers out there, Mutex lock is a thing that prevents 2 MSI packages from running at the same time. First MSI to run gets a Mutex lock and the second get MSI error 1618.

This also makes it impossible to have one MSI call another MSI.
https://social.msdn.microsoft.com/forums/windows/en-us/be861447-b6de-470f-81c0-52635e0accdc/failed-to-grab-execution-mutex-system-error-258
